modules/encoders/php/base64.rb

```diff
@@ -20,10 +20,14 @@ def initialize
   end
 
   def encode_block(state, buf)
-    # Have to have these for the decoder stub, so if they're not available,
-    # there's nothing we can do here.
+    executor = "eval"
     %w[c h r ( ) . e v a l b a s e 6 4 _ d e c o d e ;].uniq.each do |c|
-      raise BadcharError if state.badchars.include?(c)
+      if state.badchars.include?(c)
+        %w[c h r ( ) . a s s e r t b a s e 6 4 _ d e c o d e ;].uniq.each do |c|
+          raise BadcharError if state.badchars.include?(c)
+        end
+        executor = "assert"
+      end
     end
```

Comment:
This looks a bit odd. At the moment this new addition will be done for each character in the string `c h r ( ) . e v a l b a s e 6 4 _ d e c o d e ;` which I think is a bit excessive in terms of how many times it will be run.

Would it perhaps be better to set something like a "invalid_data" flag and then if that is set then run your new addition? This would ensure we aren't using extra cycles to calculate something we have already calculated the results of.

Reply:
Adding the forgotten `break` makes it better.